Add handler option

Macroz opened the pull request:
Implement handler option for Compojure based handler and live CSS reloading.

gadfly361 commented on create-templates.sh:
```diff
@@ -58,7 +58,7 @@ cd ../../..
 
 
 echo creating base +routes +re-com +test +garden
```
add `+handler` to the echo

gadfly361 commented on src/leiningen/new/re_frame/project.clj:
```diff
@@ -17,14 +19,19 @@
                                     "test/js"{{/test?}} {{#garden?}}
                                     "resources/public/css/compiled"{{/garden?}}]
 
```
Oh nice, forgot about css - maybe rewrite this section so the css-dirs is always available

```
 :figwheel {:css-dirs ["resources/public/css"] {{#handler?}}
            :ring-handler {{name}}.handler/handler{{/handler?}} }
```

Macroz replied:
Ok I did that, also added a base CSS file if garden is not used. How does that look?

gadfly361 commented:
Awesome, thanks for the PR! Made a couple comments, and after those changes we will get this merged in :)

Macroz commented:
A few questions came into my mind
- should the default CSS file be an option by itself and not dependent on garden?
- base +test seems to fail with missing secretary, have to check
- should there be better automatic tests? (separate branch?)
- should there be figwheel + CIDER support? (separate branch?)

gadfly361 commented:
For context, here are a couple of things that I am hoping to preserve:
1. Keeping the template lightweight and intuitively composable.
2. Minimizing the code that an end user would want to 'undo' or 'delete'.
3. Keeping it relatively easy for contributors to add new options.

---*should the default CSS file be an option by itself and not dependent on garden?*---
I think we should avoid having a default css file or an option for vanilla css.  A permanent default file (at least one with content) will break number 2, because many people will want to delete the contents of the file, and an option for vanilla css could get confusing.  I say confusing because I anticipate adding `+sass` and `+less` options alongside `+garden` .. and having a `+css` option as well could make things more opaque (breaking number 1).

---*should there be better automatic tests? (separate branch?)*---
(Note: assuming you mean for re-frame-template itself.) Absolutely! The current script was to make it easy for contributors to add new options (regarding number 3), but it is a limited way of 'testing' the options. If you have any ideas, definitely interested in a PR! :)

---*should there be figwheel + CIDER support? (separate branch?)*---
Adding a `+cider` option sounds reasonably to me, and I'd be interested in a PR.

---*should there be index-page function with hiccup instead of index.html (separate branch?)*---
I don't think we can get away with not having an index.html entirely, because of how expected/approachable it is.

There could be an option that switches from the html version to hiccup.  On one hand, I think it would be great to support this, because it is a use case that a number of people would have.  However, on the other hand, I think this would take a toll on number 3, because it would effectively duplicate any logic concerning the index. If someone wants to add a new option, they will have to update the hiccup version and the html version.  Perhaps the viability of doing this is dependent on if there are strides forward in the automatic tests department mentioned earlier.

-- 

Regarding this current `+handler` PR:

- remove the default css file and associated logic (if we plan to add anything regarding this, let's get a new issue and/or PR going).
- squash commits.

Macroz commented:
Alright, good comments! We can continue discussion if I get to other pull requests.

I will see about the Hiccup version sometime, because I definitely like to use it for templating in my projects. Hiccup could be used in the template to generate the index.html, if desired, to remove some duplication.

Also I will see about CIDER at least. I definitely want to start my projects with support for it. 

As for the automated template tests, I will see if I have the time. I do a lot of that sort of stuff at work.

I removed the default CSS commit and squashed everything else.

gadfly361 commented:
Thanks for the PR, I really appreciate the help! Any continued conversation/issues/PRs are more than welcomed :)
